tilelink/src/utils/OH1.scala

```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2016-2017 SiFive, Inc.
+
+package utils
+
+import chisel3._
+import chisel3.util.{Cat, OHToUInt}
+
+object OH1 {
```

Comment:
Consider upstream this util to Chisel, since it's used everywhere in RC

Reply:
I'm not sure how to convince the upstream how this encoding can be useful in a more general sense. rocket-chip used it mostly to exploit the fact that `size` in tilelink is encoded as `log2(number of bytes)` to generate counters.

Reply:
Basically, `UIntToOH1` is `UIntToOH` + `scanRightOr` using dynamic shifting, an example usage is find first one. I'm not so sure about the circuit performance here. Just keep this and wait for implementation result.